### PR TITLE
iwd (inet wireless daemon) support

### PR DESCRIPTION
**Pull request by lrusak**
This is very much a WIP. However it does work and it does work well.

The main caveat is that this relies on the kernel mac80211 framework, so out of tree drivers won't work (and some intree drivers that don't use the mac80211 framework).

I am just making this PR to get some visibility if people want to try and use it.

For those that don't know about it there was a great presentation at ELC about it.
https://youtu.be/yUxpm8ucQB8?list=PLbzoR-pLrL6pSlkQDW7RpnNLuxPq6WVUR

### CODE REVIEW

**Comment by MilhouseVH** [comment]
@lrusak can you rebase the connman commit?

**Comment by MilhouseVH** [comment]
Hmm and the kernel options. I'd like to test this on top of 4.11.0, ideally. I'll try and rebase locally so no rush...

File: projects/Generic/linux/linux.x86_64.conf [hunk 1]
```diff
@@ -5018,16 +5041,25 @@ CONFIG_CRYPTO_DRBG_HMAC=y
 # CONFIG_CRYPTO_DRBG_CTR is not set
 CONFIG_CRYPTO_DRBG=m
 CONFIG_CRYPTO_JITTERENTROPY=m
-# CONFIG_CRYPTO_USER_API_HASH is not set
-# CONFIG_CRYPTO_USER_API_SKCIPHER is not set
+CONFIG_CRYPTO_USER_API=y
+CONFIG_CRYPTO_USER_API_HASH=y
+CONFIG_CRYPTO_USER_API_SKCIPHER=y
 # CONFIG_CRYPTO_USER_API_RNG is not set
 # CONFIG_CRYPTO_USER_API_AEAD is not set
```
Comment (MilhouseVH) [inline projects/Generic/linux/linux.x86_64.conf]: Any particular reason why you left `CONFIG_CRYPTO_USER_API_RNG` and `CONFIG_CRYPTO_USER_API_AEAD` not set in Generic, but have enabled both options for RPi and RPi2?

**Comment by lrusak** [comment]
I've now updated this and tested on RPi3. I'm having issues with autoconnect but otherwise it seems to be working well.

I've split off the out of tree driver commits as I'm still not 100% sure what is compatible or not and that can be determined in the future.

File: packages/network/iwd/package.mk [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
+
+PKG_NAME="iwd"
+PKG_VERSION="0.14"
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE="https://git.kernel.org/cgit/network/wireless/iwd.git/about/"
+PKG_URL="https://www.kernel.org/pub/linux/network/wireless/iwd-${PKG_VERSION}.tar.xz"
+PKG_DEPENDS_TARGET="toolchain readline dbus"
+PKG_SECTION="network"
```
Comment (MilhouseVH) [inline packages/network/iwd/package.mk]: Small nit: `PKG_SECTION`, `PKG_ARCH` and `PKG_SHORTDESC` can all be removed.

**Comment by MilhouseVH** [comment]
I ran `make oldconfig` for Generic (using current master kernel 4.19.17) when using this PR and a number of the config options being set by this PR are being changed/removed - here is the `make oldconfig` diff against this PR: http://ix.io/1BlO

Any ideas why there are differences, did you run `make oldconfig` with additional config options that are not in this PR?

RPi/RPi2 showed no differences so they look OK!

**Comment by MilhouseVH** [comment]
Ah, did you base this PR on 4.20/5.0?

**Comment by lrusak** [comment]
@MilhouseVH updated

**Comment by lrusak** [comment]
Rebased, dropped the kernel config changes, and made iwd optional. Please merge so more people can start actively testing. The default is still to use wpa_supplicant.

This will still require kernel config changes but I'll do those separately as to not hold this 2 year old PR up anymore

**Pull request merged**